CMSgov/mint-app
# [EASI-2601, EASI-2898] - Only create solution if needed, use query param instead of location state

# EASI-2601
# EASI-2898

## Changes and Description

- Added condition to only create a solution from a possible solution if `needed` = `true`
- Added loading state/render to solution selections
- Added conditional check for null on search query
- Replaced location state with query strings
- Tweaked cypress tests for consistency



## How to test this change

Select solutions and verify only solutions that are needed or have already been created are updated by mutations

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
